fri/src/folder.rs

```diff
@@ -0,0 +1,26 @@
+use ark_ff::FftField;
+use std::sync::Arc;
+
+pub struct MultiplicativeFriFolder;
+
+#[allow(dead_code)]
+impl MultiplicativeFriFolder {
+    pub fn next_layer_element_from_two_previous_layer_elements<F: FftField>(
+        f_x: &F,
+        f_minus_x: &F,
+        eval_point: &F,
+        x: &F,
+    ) -> F {
+        let x_inv = x.inverse().unwrap();
+        Self::fold(f_x, f_minus_x, eval_point, &x_inv)
+    }
+
+    fn fold<F: FftField>(f_x: &F, f_minus_x: &F, eval_point: &F, x_inv: &F) -> F {
+        *f_x + *f_minus_x + *eval_point * (*f_x - *f_minus_x) * *x_inv
```

Comment:
Maybe add comment about what this does: it is interpolating a line through (x, f(x)) and (-x, f(-x)) then evaluating it at "eval_point"

Reply:
You can see this by setting eval_point = -x and x respectively :)